src/Storage/Imap.php

```diff
@@ -372,16 +372,31 @@ public function getFolders($rootFolder = null)
      * @throws Exception\RuntimeException
      * @throws Protocol\Exception\RuntimeException
      */
-    public function selectFolder($globalName)
+    public function selectFolder($globalName, $readOnly = false)
```

Comment:
Potential BC break for any extensions to this class, but only if those extensions are typehinting the additional, optional argument. Will need to draw attention to this in the release notes.